### PR TITLE
Add support for Entrez 0 and symbol 'Unknown' in MAF files

### PR DESCRIPTION
**Pull request by fedde-s**
Interpret the combination as indicating an intergenic mutation, and otherwise parse them as missing values. Handle these missing values the same as in other file types with these columns—fall back to the symbol if the Entrez ID is missing.

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java [hunk 1]
```diff
@@ -251,38 +251,69 @@ public void importData() throws IOException, DaoException {
...
+                String geneSymbol = record.getHugoGeneSymbol();
+                String entrezIdString = record.getGivenEntrezGeneId();
+
+                CanonicalGene gene = null;
+                if (!(entrezIdString.isEmpty() ||
+                      entrezIdString.equals("0"))) {
+                    Long entrezGeneId;
+                    try {
+                        entrezGeneId = Long.parseLong(entrezIdString);
+                    } catch (NumberFormatException e) {
```
Comment (pieterlukasse) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java]: this is not correct. We agreed to block negative values as well (see #1013)

**Comment by aderidder** [comment]
Test data used for testing:
[1142Test.tar.gz](https://github.com/cBioPortal/cbioportal/files/259028/1142Test.tar.gz)

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java [hunk 2]
```diff
@@ -251,38 +251,69 @@ public void importData() throws IOException, DaoException {
...
+                // what the column Variant_Classification says
+                if (geneSymbol.equals("Unknown") &&
+                        entrezIdString.equals("0") &&
+                        mutationType != null &&
+                        !mutationType.equalsIgnoreCase("IGR")) {
+                    ProgressMonitor.logWarning(
+                            "Treating mutation with gene symbol 'Unknown' " +
+                            "and Entrez gene ID 0 as intergenic ('IGR') " +
+                            "instead of '" + mutationType + "'.");
+                    mutationType = "IGR";
```
Comment (pieterlukasse) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java]: let's do something like 
```
myMutationFilter.igrRejects++;
entriesSkipped++;
continue;
```
and keep `if` below simple (just `if (gene == null)` )

**Comment by pieterlukasse** [comment]
@n1zea144 , @zheins : I reviewed this one. I will do some tests and update it here. If you can approve this code and tests are OK, we can merge this one.

**Comment by pieterlukasse** [comment]
tested with extended set from @aderidder and all seems fine :+1:

**Comment by zheins** [comment]
I just tested it, looks good! :+1:

**Pull request merged**